src/runtime/contrib/tensorrt/tensorrt_module.cc

```diff
@@ -197,7 +200,19 @@ class TensorRTModule : public runtime::ModuleNode {
       if (!runtime::TypeMatch(arg->dtype, kDLFloat, 32)) {
         LOG(FATAL) << "Only float32 inputs are supported.";
       }
-      bindings[binding_index] = reinterpret_cast<float*>(arg->data);
+      if (inputs[i]->ctx.device_type == kDLCPU) {
```

Comment:
I think we should reverse this logic:
```
if (device == GPU) {
  // pass pointer directly
} else {
  // (maybe allocate gpu buffer)
  // move data from whatever ctx it came from to gpu
}
```

Reply:
I have made this change to reverse the logic of checking device_type.